st2client/st2client/commands/auth.py

```diff
@@ -70,6 +73,84 @@ def run_and_print(self, args, **kwargs):
                               attributes=self.display_attributes, json=args.json, yaml=args.yaml)
 
 
+class LoginCommand(resource.ResourceCommand):
+    display_attributes = ['user', 'token', 'expiry']
+
+    def __init__(self, resource, *args, **kwargs):
+
+        kwargs['has_token_opt'] = False
+
+        super(LoginCommand, self).__init__(
+            resource, kwargs.pop('name', 'create'),
+            'Authenticate user, acquire access token, and update CLI config directory',
+            *args, **kwargs)
+
+        self.parser.add_argument('username',
+                                 help='Name of the user to authenticate.')
+
+        self.parser.add_argument('-p', '--password', dest='password',
+                                 help='Password for the user. If password is not provided, '
+                                      'it will be prompted.')
+        self.parser.add_argument('-l', '--ttl', type=int, dest='ttl', default=None,
+                                 help='The life span of the token in seconds. '
+                                      'Max TTL configured by the admin supersedes this.')
+        self.parser.add_argument('-w', '--write-real-password', action='store_true', default=False,
+                                 dest='write_password',
+                                 help='Write the real (plain-text) password to the config file '
+                                      '(default is to omit it')
+
+    def run(self, args, **kwargs):
+
+        if not args.password:
+            args.password = getpass.getpass()
+        instance = self.resource(ttl=args.ttl) if args.ttl else self.resource()
+
+        cli = BaseCLIApp()
+
+        # Determine path to config file
+        try:
+            config_file = cli._get_config_file_path(args)
+        except ValueError:
+            # config file not found in args or in env, defaulting
+            config_file = expanduser('~/.st2/config')
+
+        logging.basicConfig(
+            level=logging.CRITICAL,
+            handlers=[logging.NullHandler]
+        )
+        cli.LOG = logging.getLogger("st2client.base")
```

Comment:
FWIW, here's the error I get when I don't set up logging. It seemed strange that the log was being defined on the instance and not with the typical `LOG = logging.getLogger(__name__)` but figured it was for some reason.

```
(virtualenv)vagrant@st2dev:~/st2/st2client$ python setup.py install && st2 --debug login st2admin -p Password1!
# -------- begin 140500329551632 request ----------
curl -X POST -H  'Connection: keep-alive' -H  'Accept-Encoding: gzip, deflate' -H  'Accept: */*' -H  'User-Agent: python-requests/2.11.1' -H  'content-type: application/json' -H  'Content-Length: 2' -H  'Authorization: Basic c3QyYWRtaW46UGFzc3dvcmQxIQ==' --data-binary '{}' http://127.0.0.1:9100/tokens
# -------- begin 140500329551632 response ----------
{
    "user": "st2admin",
    "token": "11d044175f514b7dbd5b3b15557f09fe",
    "expiry": "2016-12-21T18:37:53.506004Z",
    "id": "58597a8132ed3534e5d0a586",
    "metadata": {}
}
# -------- end 140500329551632 response ------------

Failed to log in as st2admin
ERROR: 'NoneType' object has no attribute 'debug'

CLI settings:
----------------
Config file path: /home/vagrant/.st2/config
Client settings:
----------------
ST2_BASE_URL: http://127.0.0.1
ST2_AUTH_URL: http://127.0.0.1:9100
ST2_API_URL: http://127.0.0.1:9101/v1
ST2_AUTH_TOKEN: None

Proxy settings:
---------------
HTTP_PROXY:
HTTPS_PROXY:

Traceback (most recent call last):
  File "/home/vagrant/st2/virtualenv/lib/python2.7/site-packages/st2client-2.2.dev0-py2.7.egg/st2client/shell.py", line 282, in run
    args.func(args)
  File "/home/vagrant/st2/virtualenv/lib/python2.7/site-packages/st2client-2.2.dev0-py2.7.egg/st2client/commands/auth.py", line 146, in run_and_print
    self.run(args, **kwargs)
  File "/home/vagrant/st2/virtualenv/lib/python2.7/site-packages/st2client-2.2.dev0-py2.7.egg/st2client/commands/auth.py", line 125, in run
    cli._cache_auth_token(token_obj=manager)
  File "/home/vagrant/st2/virtualenv/lib/python2.7/site-packages/st2client-2.2.dev0-py2.7.egg/st2client/base.py", line 321, in _cache_auth_token
    self.LOG.debug('Token has been cached in "%s"' % (cached_token_path))
AttributeError: 'NoneType' object has no attribute 'debug'
```